### PR TITLE
Instantiate Mock members by function

### LINKED ISSUE #516: "500: Internal Server Error - Failed to stop mission" in the tests
**Describe the bug**
500: Internal Server Error - Failed to stop mission" in the tests

**To Reproduce**
run `pytest .`

**Expected behavior**

**Screenshots**
```
FAILED tests/isar/state_machine/test_state_machine.py::test_state_machine_with_successful_collection - assert 0 == 1
FAILED tests/isar/state_machine/test_state_machine.py::test_state_machine_with_unsuccessful_collection - AssertionError: assert deque([idle]) == deque([idle, ...itiate, idle])
FAILED tests/isar/state_machine/test_state_machine.py::test_state_machine_with_successful_mission_stop - fastapi.exceptions.HTTPException: 500: Internal Server Error - Failed to stop mission
FAILED tests/isar/state_machine/test_state_machine.py::test_state_machine_with_unsuccessful_mission_stop - fastapi.exceptions.HTTPException: 500: Internal Server Error - Failed to stop mission
```

### PR DESCRIPTION
**Pull request by sondreo**
Fixes bug in state machine tests due to class instance member being mutated across different tests.

Resolves #516

### CODE REVIEW

File: tests/mocks/step.py [hunk 1]
```diff
@@ -5,7 +5,10 @@
 
 
 class MockStep:
```
Comment (tsundvoll) [inline tests/mocks/step.py:7]:
```suggestion
class DummyStep:
```

Also consider using the naming "Stub" or "Fake", both may be representative of what we try to accomplish here. Relevant article: https://jesusvalerareales.com/testing-with-test-doubles/

Also, does not need to be solved here, you can make an issue for it and suggest a renaming for all similar "Mocks" that should be called "Dummy", "Stub", or "Fake"

**Review by tsundvoll — CHANGES_REQUESTED** [review]
Well spotted that you found the cause of the issue. Some comments for naming:

File: tests/mocks/step.py (hunk 1 above)
Comment (sondreo) [reply tests/mocks/step.py:7]: Good idea, and great tip for a relevant resource. I will defer this to a separate issue, and have [created one](https://github.com/equinor/isar/issues/524).

File: tests/mocks/step.py [hunk 2]
```diff
@@ -5,7 +5,10 @@
...
-    drive_to = DriveToPose(pose=MockPose.default_pose)
-    take_image_in_coordinate_direction = TakeImage(
-        target=Position(x=1, y=1, z=1, frame=Frame("robot"))
-    )
+    @staticmethod
+    def drive_to() -> DriveToPose:
+        return DriveToPose(pose=MockPose.default_pose)
```
[the remark this answers is not in the document]
Comment (sondreo) [reply tests/mocks/step.py]: Good point, I have extended the PR with an additional commit addressing this, for consistency and uniformity in the Mock implementation.

File: tests/mocks/step.py [hunk 3]
```diff
@@ -5,7 +5,10 @@
...
+
+    @staticmethod
+    def take_image_in_coordinate_direction() -> TakeImage:
```
[the remark this answers is not in the document]
Comment (sondreo) [reply tests/mocks/step.py:13]: See below regarding naming. It is added as a separate issue.

**Review by tsundvoll — APPROVED** [review]
Good work 👍  Wise to delegate comments to a separate issue

**Pull request merged**